File: RQ3/C-REval-Desc/task23_desc.c
int largest_prime_factor(int n) {
    int largest = 1;
    for (int j = 2; j <= n; j++) {
        int is_prime = 1;
        if (j < 2) {
            is_prime = 0;
        }
        for (int i = 2; i < j - 1; i++) {
            if (j % i == 0) {
                is_prime = 0;
            }
        }
        if (n % j == 0 && is_prime) {
            largest = (largest > j) ? largest : j;
        }
    }
    int _result = largest;

    assert(func(_result));  // POST: func is equivalent to: Return the largest prime factor of n. Assume n > 1 and is not a prime.
}

File: RQ1/REval-Desc/task23_desc.py
def largest_prime_factor(n: int):
    largest = 1
    for j in range(2, n + 1):
        is_prime = True
        if j < 2:
            is_prime = False
        for i in range(2, j - 1):
            if j % i == 0:
                is_prime = False
        if n % j == 0 and is_prime:
            largest = max(largest, j)
    _result = largest

    assert func(_result)  # POST: func is equivalent to: Return the largest prime factor of n. Assume n > 1 and is not a prime. >>> largest_prime_factor(13195) 29 >>> largest_prime_factor(2048) 2
